gas-oracle/tokenprice/tokenprice_uniswap.go

```diff
@@ -0,0 +1,82 @@
+package tokenprice
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	"github.com/mantlenetworkio/mantle/gas-oracle/bindings"
+)
+
+var (
+	bit2wETHPairAddress  = common.HexToAddress("0x5c128d25a21f681e678cb050e551a895c9309945")
+	usdt2wETHPairAddress = common.HexToAddress("0x11b815efB8f581194ae79006d24E0d814B7697F6")
+
+	wETHAddress     = common.HexToAddress("0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2")
+	bitTokenAddress = common.HexToAddress("0x1A4b46696b2bB4794Eb3D4c26f1c55F9170fa4C5")
+	usdtAddress     = common.HexToAddress("0xdAC17F958D2ee523a2206206994597C13D831ec7")
+
+	bitTokenDecimals = floatStringToBigInt("1", 18)
+	usdtDecimals     = floatStringToBigInt("1", 6)
+
+	ContractV3Quoter = common.HexToAddress("0xb27308f9F90D607463bb33eA1BeBb41C27CE5AB6")
+)
+
+type uniswapClient struct {
+	uniswapV3Quoter *bindings.Uniswapv3Quoter
+}
+
+func newUniswapClient(uniswapURL string) (*uniswapClient, error) {
+	l1Client, err := ethclient.Dial(uniswapURL)
+	if err != nil {
+		return nil, err
+	}
+
+	uniswapV3Quoter, err := bindings.NewUniswapv3Quoter(ContractV3Quoter, l1Client)
```

Comment:
How can we upgrade to the new interface after Uniswap upgrades to v4?

Reply:
> How can we upgrade to the new interface after Uniswap upgrades to v4?

The implementation method of Uniswap v4 is not yet determined. We will look at the solution after it is confirmed later.